### ISSUE TITLE
Regular expression match (?:XX|X)* dumps core

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#133]
Migrated from [rt.perl.org#938](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=938) (status was 'resolved')

Searchable as RT938$

**Comment by p5pRT** [comment]
### From garethr@cre.canon.co.uk



The following program causes a segmentation violation&#8203;:

&nbsp;    $\_ = 'X' x 50000;
&nbsp;    /\(?&#8203;:XX|X\)\*/;

<details><summary>Perl Info</summary>

```


Site configuration information for perl 5.00502:

Configured by abw at Thu Nov 19 10:50:08 GMT 1998.

Summary of my perl5 (5.0 patchlevel 5 subversion 2) configuration:
  Platform:
    osname=solaris, osvers=2.6, archname=sun4-solaris
    uname='sunos bandanna 5.6 generic_105181-03 sun4u sparc sunw,ultra-1 '
    hint=previous, useposix=true, d_sigaction=define
    usethreads=undef useperlio=undef d_sfio=undef
  Compiler:
    cc='gcc', optimize='-O', gccversion=2.8.0
    cppflags='-I/opt/gnu/include -I/user/perl/build/include'
    ccflags ='-I/opt/gnu/include -I/user/perl/build/include'
    stdchar='unsigned char', d_stdstdio=define, usevfork=false
    intsize=4, longsize=4, ptrsize=4, doublesize=8
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    alignbytes=8, usemymalloc=n, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags ='-L/opt/gnu/lib -L/user/perl/build/lib'
    libpth=/opt/gnu/lib /lib /usr/lib /usr/ccs/lib /user/perl/build/lib
    libs=-lsocket -lnsl -lgdbm -ldl -lm -lc -lcrypt -ldb
    libc=/lib/libc.so, so=so, useshrplib=false, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags=' '
    cccdlflags='-fPIC', lddlflags='-G -L/opt/gnu/lib -L/user/perl/build/lib'

Locally applied patches:
    


@INC for perl 5.00502:
    /apps/perl5/lib/5.00502/sun4-solaris
    /apps/perl5/lib/5.00502
    /apps/perl5/lib/site_perl/5.005/sun4-solaris
    /apps/perl5/lib/site_perl/5.005
    .


Environment for perl 5.00502:
    HOME=/user/garethr
    LANG=C
    LD_LIBRARY_PATH=/usr/dt/lib:/usr/openwin/lib:/apps/X11R5/lib:/usr/lib
    LOGDIR (unset)
    PATH=/usr/dt/bin:/usr/openwin/bin:/bin:/usr/bin:/usr/ucb:/bin:/usr/bin:/cre/bin:/apps/gcc-2.8.0/bin:/apps/frame5/bin:/apps/emacs/bin:/apps/tex/bin:/apps/perl5/bin:/user/web/bin/scripts:/user/web/bin/SunOS-5.6:/apps/gnu/bin:/apps/mysql/bin:/usr/ccs/bin:/apps/X11R5/bin:/usr/openwin/bin
    PERL_BADLANG (unset)
    SHELL=/cre/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

This regexp bug can \_still\_ cause a segfault in bleadperl as of today\.

\-\-cut\-\-
\#\!/usr/local/bin/perl

$\_ = 'X' x 50000;
/\(?&#8203;:XX|X\)\*/;
\-\-cut\-\-

**Comment by p5pRT** [comment]
### From @vanstyn

In \<20000918115335\.A27376@&#8203;athenaeum\.globnix\.org>\, Michael Stevens writes&#8203;:
:This regexp bug can \_still\_ cause a segfault in bleadperl as of today\.
:
:\-\-cut\-\-
:\#\!/usr/local/bin/perl
:
:$\_ = 'X' x 50000;
:/\(?&#8203;:XX|X\)\*/;

Yes\, this is not likely to be fixed any time soon&#8203;: the recursion exceeds
your stack size\. Increasing the stack size for perl may allow you to
match longer strings before it breaks\, but a fix involves flattening
the recursion out of the regexp matcher\, which is a\) difficult and b\)
not of the highest priority\. But patches welcome\, of course\. :\)

Hugo

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

On Mon\, Sep 18\, 2000 at 01&#8203;:33&#8203;:17PM \+0100\, Hugo wrote&#8203;:
> Yes\, this is not likely to be fixed any time soon&#8203;: the recursion exceeds
> your stack size\. Increasing the stack size for perl may allow you to
> match longer strings before it breaks\, but a fix involves flattening
> the recursion out of the regexp matcher\, which is a\) difficult and b\)
> not of the highest priority\. But patches welcome\, of course\. :\)

It would be nice if it could at least say "this is too big\, I give up"\.

Code wise\, the regexp stuff scares me\.

Michael\.

**Comment by p5pRT** [comment]
### From @vanstyn

In \<20000918141050\.A29064@&#8203;athenaeum\.globnix\.org>\, Michael Stevens writes&#8203;:
:On Mon\, Sep 18\, 2000 at 01&#8203;:33&#8203;:17PM \+0100\, Hugo wrote&#8203;:
:> Yes\, this is not likely to be fixed any time soon&#8203;: the recursion exceeds
:> your stack size\. Increasing the stack size for perl may allow you to
:> match longer strings before it breaks\, but a fix involves flattening
:> the recursion out of the regexp matcher\, which is a\) difficult and b\)
:> not of the highest priority\. But patches welcome\, of course\. :\)
:
:It would be nice if it could at least say "this is too big\, I give up"\.
:
:Code wise\, the regexp stuff scares me\.

It should do&#8203;: that is what the REG\_INFTY stuff is there for\. However
it is difficult \(impossible?\) for Configure to determine the right
value to set REG\_INFTY to \- I believe it is usually too high under
Linux\, for example \- and I have seen suggestions that the REG\_INFTY
check may not be triggered when it should in all cases\.

HTH\,

Hugo

**Comment by p5pRT** [comment]
### From The RT System itself

still core dumps in bleadperl as of 18/sep/2000\, perl 5\.7\.0 DEVEL7093
